src/gui/dialogs/window_dlg_wait.c

```diff
@@ -105,39 +103,57 @@ void window_dlg_wait_draw(window_dlg_wait_t *window) {
                 display->draw_line(point_ui16(x + 6, y + 32), point_ui16(x + 24 - 1, y + 32), COLOR_ORANGE);
                 display->draw_line(point_ui16(x + 6, y + 33), point_ui16(x + 24 - 1, y + 33), COLOR_ORANGE);
             }
-        } else
-            render_icon_align(icon_rc, IDR_PNG_wizard_icon_hourglass, COLOR_BLACK, ALIGN_CENTER); //change icon
-
-        window->flags &= 0xF00F; //reset animation flags
+        }
     }
 
-    if (window->flags & DLG_W8_PROGRESS) { //progress changed
-        rect_ui16_t rc_pro = rc;
-        rc_pro.x = 10;
-        rc_pro.w -= 20;
-        const unsigned int text_len = 16;
-        char text[text_len];
-        rc_pro.h = 16;
-        rc_pro.y += 120;
-        const uint16_t w = rc_pro.w;
-        rc_pro.w = w * window->progress / 100;
-        display->fill_rect(rc_pro, COLOR_ORANGE);
-        rc_pro.x += rc_pro.w;
-        rc_pro.w = w - rc_pro.w;
-        display->fill_rect(rc_pro, COLOR_GRAY);
-        rc_pro.y += rc_pro.h;
-        rc_pro.w = rc.w - 120;
-        rc_pro.x = rc.x + 60;
-        rc_pro.h = 30;
-        snprintf(text, text_len, "%d%%", window->progress);
-        render_text_align(rc_pro, text, window->font_title, window->color_back, window->color_text, window->padding, ALIGN_CENTER);
-        window->flags &= 0x7FFF;
+    if (window->components & DLG_W8_DRAW_PROGRESS) { // progress bar enabled
+        if (window->progress_chng) {                 // progress changed
+            window->progress_chng = false;
+
+            char text[8];
+            rect_ui16_t rc_pro = rc;
+            rc_pro.x = 10;
+            rc_pro.w -= 20;
+            rc_pro.h = 16;
+            rc_pro.y += 120;
+            display->fill_rect(rc_pro, COLOR_GRAY);
+
+            if (window->progress != -1) {
+                const uint16_t w = rc_pro.w;
+                rc_pro.w = (w * window->progress) / 100;
+                display->fill_rect(rc_pro, COLOR_ORANGE);
+                rc_pro.x += rc_pro.w;
+                rc_pro.w = w - rc_pro.w;
+                sprintf(text, "%d%%", window->progress);
+            } else {
+                strcpy(text, "N/A");
```

Comment:
strlcpy!!!! how many times shall I say that?

Reply:
commit `0ace17e`